### DISCUSSION TITLE
Reports not visible in UI until imported via UI

### DISCUSSION THREAD

**Discussion by winadm87** [netbox-community/netbox#14278] [Help Wanted!] [answered]
ENV:
Netbox infra done by docker compose from here https://github.com/netbox-community/netbox-docker/wiki/Getting-Started

Steps:
Volume with reports mounted to netbox container 
`    volumes:
...
    - ./netbox-reports-files:/opt/netbox/netbox/reports:rw
`
Basic report is put into the mounted folder, it can be successfully visible in the container
`
unit@367bdd43b99f:/opt/netbox/netbox$ ls -al reports/
total 48
drwxr-xr-x 10 root root  320 Nov 16 06:27 .
drwxr-xr-x  1 root root 4096 Nov 16 06:28 ..
...
-rw-r--r--  1 unit root  874 Nov  3 12:39 ip_report.py
`

Expected Behavior
The report is visible in UI in http://localhost:8000/extras/reports/

Observed Behavior
The report is not visible in UI. But if next we import the same report via UI - it is imported successfully and we can manage the report directly from code and all the changes to the report are visible in the UI. We supposed that after import via UI report file will have some other access rights - but they are still the same
`
unit@367bdd43b99f:/opt/netbox/netbox$ ls -al reports/
total 48
drwxr-xr-x 10 unit root  320 Nov 16 06:27 .
drwxr-xr-x  1 root root 4096 Nov 16 06:28 ..
...
-rw-r--r--  1 unit root  874 Nov 16 06:46 ip_report.py
`

**Comment by candlerb** [comment, marked as answer]
Reports handling changed fairly recently, possibly in [v3.5.0](https://github.com/netbox-community/netbox/releases/v3.5.0). Now scripts and reports are listed in the database and must be imported via the UI; simply dropping them into the directory is not sufficient. See also #12766 and #12842.

As long as they're in a volume (or alternatively in an S3 bucket) then you should be fine: they'll persist when you upgrade the container, and you just have to remember to upload them via the UI rather than dropping them straight into the volume.

The documentation at https://github.com/netbox-community/netbox-docker/wiki/Custom-Reports-&-Scripts needs to be updated though.

**Comment by winadm87** [comment]
This behavior sounds exactly like what we thought as observed the situation. @candlerb many thanks
